multisig-wallet/project/multisig-contract/tests/functions/constructor.rs

```diff
@@ -0,0 +1,86 @@
+use crate::utils::{
+    abi_calls::{constructor, nonce},
+    test_helpers::{constructor_users, setup_env, DEFAULT_THRESHOLD},
+    User, VALID_SIGNER_PK,
+};
+
+use fuels::prelude::*;
+
+mod success {
+
+    use super::*;
+
+    #[tokio::test]
+    async fn setup_with_constructor() {
+        let (_private_key, deployer, _non_owner) = setup_env(VALID_SIGNER_PK).await.unwrap();
+
+        constructor(&deployer.contract, constructor_users(), DEFAULT_THRESHOLD).await;
+
+        assert_eq!(nonce(&deployer.contract).await.value, 1);
+    }
+}
+
+mod revert {
+
+    use super::*;
+
+    #[tokio::test]
+    #[should_panic(expected = "CannotReinitialize")]
+    async fn cannot_reinitialize() {
+        let (_private_key, deployer, _non_owner) = setup_env(VALID_SIGNER_PK).await.unwrap();
+
+        constructor(&deployer.contract, constructor_users(), DEFAULT_THRESHOLD).await;
+
+        constructor(&deployer.contract, constructor_users(), DEFAULT_THRESHOLD).await;
+    }
+
+    #[tokio::test]
+    #[should_panic(expected = "AddressCannotBeZero")]
+    async fn address_cannot_be_zero() {
+        let (_private_key, deployer, _non_owner) = setup_env(VALID_SIGNER_PK).await.unwrap();
+
+        let mut users = constructor_users();
+        users[0] = User {
+            address: Bits256::from_hex_str(
+                "0x0000000000000000000000000000000000000000000000000000000000000000",
+            )
+            .unwrap(),
+            weight: 3,
+        };
+
+        constructor(&deployer.contract, users, DEFAULT_THRESHOLD).await;
+    }
+
+    #[tokio::test]
+    #[should_panic(expected = "ThresholdCannotBeZero")]
+    async fn threshold_cannot_be_zero() {
```

Comment:
I would order the tests to be in the same order as the requirements so that it's easier to see requirement by requirement which test should be tested and thus if anything is missing